lib/Backend/arm64/LowerMD.cpp

```diff
@@ -6008,73 +5958,70 @@ LowererMD::GenerateClz(IR::Instr * instr)
 void
 LowererMD::SaveDoubleToVar(IR::RegOpnd * dstOpnd, IR::RegOpnd *opndFloat, IR::Instr *instrOrig, IR::Instr *instrInsert, bool isHelper)
 {
+    Assert(opndFloat->GetType() == TyFloat64);
+
     // Call JSNumber::ToVar to save the float operand to the result of the original (var) instruction
-    IR::Opnd * symVTableDst;
-    IR::Opnd * symDblDst;
-    IR::Opnd * symTypeDst;
-    IR::Instr *newInstr;
-    IR::Instr * numberInitInsertInstr = nullptr;
-    if (instrOrig->dstIsTempNumber)
-    {
-        // Use the original dst to get the temp number sym
-        StackSym * tempNumberSym = this->m_lowerer->GetTempNumberSym(instrOrig->GetDst(), instrOrig->dstIsTempNumberTransferred);
-
-        // LEA dst, &tempSym
-        IR::SymOpnd * symTempSrc = IR::SymOpnd::New(tempNumberSym, TyMachPtr, this->m_func);
-        newInstr = IR::Instr::New(Js::OpCode::LEA, dstOpnd, symTempSrc, this->m_func);
+
+    // s1 = MOV opndFloat
+    IR::RegOpnd *s1 = IR::RegOpnd::New(TyMachReg, m_func);
+    IR::Instr *mov = IR::Instr::New(Js::OpCode::FMOV, s1, opndFloat, m_func);
+    instrInsert->InsertBefore(mov);
```

Comment:
You need to use FMOV_GEN to move from a floating point register to an integer register.